python/ray/actor.py

```diff
@@ -214,6 +225,12 @@ def __ray_terminate__(self, actor_id):
             # remove the actor key from Redis here.
             ray.worker.global_worker.redis_client.hset(b"Actor:" + actor_id,
                                                        "removed", True)
+            # Release the GPUs that this worker was using.
+            if len(ray.get_gpu_ids()) > 0:
+                release_gpus_in_use(self.__ray_driver_id__,
+                                    self.__ray_local_scheduler_id__,
+                                    ray.get_gpu_ids(),
+                                    ray.worker.global_worker.redis_client)
```

Comment:
If I'm understanding this correctly, isn't this being called from the actor process, so potentially a failure before this point could cause the resources to get leaked indefinitely?

Reply:
A failure before this point could cause the resources to get leaked from the point of the failure until the end of the "job". That is, when a driver exits, all of the GPUs used by any of its actors are released.

Ultimately we do want to get rid of this code path and have just the local scheduler release the resources.